src/inventory_ui.cpp

```diff
@@ -944,63 +1125,23 @@ item_location game::inv_map_splice(
                             slices.back().emplace_back( &current_stack.back(), INT_MIN );
                             opts.emplace( &current_stack.back().front(), item_location( vehicle_cursor( *veh, part ), &it ) );
 
-                            if( cur_invlet <= max_invlet ) {
-                                current_stack.back().front().invlet = cur_invlet++;
-                                invlets.emplace_back( vehicle_cursor( *veh, part ), &it );
-                            } else {
-                                current_stack.back().front().invlet = 0;
-                            }
+                            current_stack.back().front().invlet = ( cur_invlet <= max_invlet ) ? cur_invlet++ : 0;
                         }
                     }
                 }
                 std::string name = trim( std::string( _( "VEHICLE" ) )  + " " + direction_suffix( g->u.pos(), pos ) );
                 categories.emplace_back( name, name, rank-- );
-                inv_s.make_item_list( slices.back(), &categories.back() );
+                inv_s.add_items( slices.back(), inventory_selector::AT_END, &categories.back() );
             }
         }
     }
 
-    inv_s.prepare_paging();
-
-    while( true ) {
-        inv_s.display();
-        const std::string action = inv_s.ctxt.handle_input();
-        const long ch = inv_s.ctxt.get_raw_input().get_first_input();
-        const int item_pos = g->u.invlet_to_position( ch );
-
-        if( item_pos != INT_MIN ) {
-            // Indexed item in inventory
-            inv_s.set_to_drop( item_pos, 0 );
-            return item_location( u, inv_s.first_item );
-
-        } else if( ch >= min_invlet && ch <= max_invlet ) {
-            // Indexed item on ground or in vehicle
-            if( (long)invlets.size() > ch - min_invlet ) {
-                return std::move( invlets[ch - min_invlet] );
-            }
-
-        } else if( inv_s.handle_movement( action ) ) {
-            // continue with comparison below
-
-        } else if( action == "QUIT" ) {
-            return item_location();
-
-        } else if( action == "RIGHT" || action == "CONFIRM" ) {
-            inv_s.set_selected_to_drop( 0 );
-
-            // Item in inventory
-            if( inv_s.get_selected_item_position() != INT_MIN ) {
-                return item_location( u, inv_s.first_item );
-            }
-            // Item on ground or in vehicle
-            auto it = opts.find( inv_s.first_item );
-            if( it != opts.end() ) {
-                return std::move( it->second );
-            }
-
-            return item_location();
-        }
+    if( inv_s.empty() ) {
+        const std::string msg = ( none_message.empty() ) ? _( "You don't have the necessary item at hand." ) : none_message;
+        popup_getkey( msg.c_str() );
```

Comment:
Either use the overload of `popup` that accepts a `std::string` (but it requires a explicit flag value), or use `"%s"` as format. The popup functions that accept a C-String do formating via `printf`, if the message contains a format specifier (like "%s"), you get undefined behavior.

One scenario to get the format specifier is if someone names their toolkit item "The %s master" and tries to use it, `none_message` would contain the "%s".)

This applies to all calls to popup with a user supplied C-String argument, using a C-String literal is usually fine.